app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  # Method that updates the posts counter for a user.
+
+  def update_posts_counter
+    author.update(post_counter: author.posts.count)
+  end
```

Comment:
**Nice job creating the different counter update method correctly**
### Additional Support

- [ ] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

- [ ] Please do the same for `after_save :update_comments_counter` and `after_save :update_likes_counter`

```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```